### PR TITLE
New package: Microsoft.UrlRewrite version 2.1

### LINKED ISSUE #184166: [Package Request]: URL Rewrite for IIS
### How can we help?

I would like someone else to build the manifest.

### Please read and ensure the following

- [x] The installer meets the above requirements

### Please provide the following information

Download Page Url: https://www.iis.net/downloads/microsoft/url-rewrite
Publisher: Microsoft
Package Name: URL Rewrite for IIS
Description: IIS URL Rewrite 2.1 enables Web administrators to create powerful rules to implement URLs that are easier for users to remember and easier for search engines to find. By using rule templates, rewrite maps, .NET providers, and other functionality integrated into IIS Manager, Web administrators can easily set up rules to define URL rewriting behavior based on HTTP headers, HTTP response or request headers, IIS server variables, and even complex programmatic rules. In addition, Web administrators can perform redirects, send custom responses, or stop HTTP requests based on the logic expressed in the rewrite rules.
Package Version: 2.1
Installer Url: https://download.microsoft.com/download/1/2/8/128E2E22-C1B9-44A4-BE2A-5859ED1D4592/rewrite_amd64_en-US.msi

### PR DESCRIPTION
**Pull request by Dragon1573**
## Checklist for Pull Requests

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Resolve #184166

## Manifests

- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the directory's name containing the manifest you're submitting.

## Additional information

> [!WARNING]
>
> In order to install successfully, **IIS 7.0 CoreWebEngine** and **W3SVC** functions should be already installed.

What is the package identifier or Windows Feature name of them? 😕

image
image

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/184181)

### CODE REVIEW

**Comment by stephengillie** [comment]
Manual Validation ended with:
> 5:20:30 PM: 
> 2024-10-21 17:20:20.832 [CORE] Failed to enable Windows Feature [IIS-WebServer] with exit code: 50

(Automated response - build 897.)

**Comment by stephengillie** [comment]
The blocker here is another feature on which the `IIS-WebServer` feature depends. I tried adding it manually:

```PowerShell
PS C:\Users\User\Desktop> Enable-WindowsOptionalFeature -FeatureName IIS-WebServer -Online
Enable-WindowsOptionalFeature : One or several parent features are disabled so current feature can not be enabled.
At line:1 char:1
+ Enable-WindowsOptionalFeature -FeatureName IIS-WebServer -Online
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Enable-WindowsOptionalFeature], COMException
    + FullyQualifiedErrorId : Microsoft.Dism.Commands.EnableWindowsOptionalFeatureCommand

PS C:\Users\User\Desktop> Enable-WindowsOptionalFeature -FeatureName IIS-WebServer -All -Online
Enable-WindowsOptionalFeature : The referenced assembly could not be found.
At line:1 char:1
+ Enable-WindowsOptionalFeature -FeatureName IIS-WebServer -All -Online
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Enable-WindowsOptionalFeature], COMException
    + FullyQualifiedErrorId : Microsoft.Dism.Commands.EnableWindowsOptionalFeatureCommand
PS C:\Users\User\Desktop> 

```
Output of `Get-WindowsOptionalFeature -Online | Select-Object FeatureName | Where-Object {$_ -match "IIS-"}` matched against Windows Features GUI: 
image

Tried enabling manually
image

"Tell me how to solve this problem" shows it's the .NET Framework 3.5: 
image

Download web installer:
image
https://dotnet.microsoft.com/en-us/download/dotnet-framework/net35-sp1

Unfortunately, this made no change.
image

**Comment by BurnsSilver** [comment]
I have a script which install the features and on a clean install of windows running them in this order works. Let me know if that helps.

Enable-WindowsOptionalFeature -FeatureName NetFx3 -Online -NoRestart -All
Enable-WindowsOptionalFeature -FeatureName IIS-WebServerRole -Online -NoRestart -All
Enable-WindowsOptionalFeature -FeatureName IIS-WebServer -Online -NoRestart -All

File: manifests/m/Microsoft/UrlRewrite/2.1/Microsoft.UrlRewrite.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+- interactive
+- silent
+- silentWithProgress
+UpgradeBehavior: install
+InstallationMetadata:
+  DefaultInstallLocation: .\inetsrv
+Dependencies:
+  WindowsFeatures:
+  # IIS 7.0 CoreWebEngine and W3SVC should be installed and enabled.
+  - IIS-WebServer
```
Comment (Dragon1573) [inline manifests/m/Microsoft/UrlRewrite/2.1/Microsoft.UrlRewrite.installer.yaml:17]:
```suggestion
  - NetFx3
  - IIS-WebServerRole
  - IIS-WebServer
```

- See https://github.com/microsoft/winget-pkgs/pull/184181#issuecomment-2429257816

**Comment by stephengillie** [comment]
Manual Validation ended with:
> 12:35:16 PM: 
> 2024-10-24 12:35:15.088 [CORE] Failed to enable Windows Feature [IIS-WebServerRole] with exit code: 14081

Automated error analysis suggests 14081 may mean The referenced assembly could not be found. Is a dependency missing?

(Automated response - build 897.)

**Comment by Dragon1573** [comment]
> [!CAUTION]
>
> Close with reason: **Unable to resolve dependencies or Windows Features**.

**Pull request closed without merging**